### ISSUE TITLE
Move to Pandas to improve `Reader` and `FeatureSet` performance

### ISSUE THREAD

**Issue by jbiggsets** [EducationalTestingService/skll#364]
Reading data into `FeatureSet` objects using the `Reader` classes seems to take a long time, compared with simply using `pandas` IO Tools. I did some quick benchmarking using a 200 MB TSV to test the following scenarios:

- Read TSV into DataFrame using `pd.read_csv()` function. *[9 seconds]*
- Read TSV into DataFrame using `pd.read_csv()` function, and then load `pd.DataFrame` to `FeatureSet` using `FeatureSet.from_data_frame()`. *[1 minute 17 seconds]*
- Load TSV into `FeatureSet` using `Reader.read()` method of `TSVReader` object. *[2 minutes 29 seconds]*

See attached for a copy of the Jupyter notebook I used for testing ([reader_speed_tests.zip](https://github.com/EducationalTestingService/skll/files/1257755/reader_speed_tests.zip)). For the last scenario, I also used `%prun` to get a sense of which function calls were taking the most time. Below are the top five function calls, ranked by total time (in seconds):

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)

 54600000  101.372    0.000  101.372    0.000 readers.py:685(safe_float)

   200002   22.954    0.000  141.354    0.001 readers.py:515(_sub_read)

        1   19.265   19.265   98.357   98.357 dict_vectorizer.py:133(fit_transform)

   200002    8.522    0.000   15.881    0.000 csv.py:108(__next__)

   200004    6.952    0.000    7.125    0.000 {built-in method builtins.next}
```

Recommendation: Move to `pandas` IO Tools in the next release, where operations like type conversion will be vectorized. This would obviously add a new dependency to `SKLL`, and there may be other considerations, but it seems like it would be worth it, if only for performance reasons.

**Comment by desilinguist** [comment]
@jbiggsets thanks for the detailed report! I agree that it's time to start using `pandas` for reading tabular data in SKLL. It'd be weird to me if someone who wants to use `skll` would have a problem with `pandas` as a dependency since it's so pervasive now and so easily installable.

**Comment by desilinguist** [comment]
Let's keep this open.

**Comment by desilinguist** [comment]
Keep it open.

**Comment by desilinguist** [comment]
Keep it open please.

**Comment by desilinguist** [comment]
Keep it open please.

**Comment by desilinguist** [comment]
Keep it open.

**Comment by desilinguist** [comment]
This should be closed before #494 is addressed.

**Comment by desilinguist** [comment]
Addressed by combination of #475 and #518
